romancal/source_catalog/source_catalog_step.py

```diff
@@ -43,6 +43,8 @@ class SourceCatalogStep(RomanStep):
         ci2_star_threshold = float(default=1.8)  # CI 2 star threshold
         suffix = string(default='cat')        # Default suffix for output files
         fit_psf = boolean(default=True)      # fit source PSFs for accurate astrometry
+        return_updated_model = boolean(default=False)   # should the image model be updated and returned
```

Comment:
Why not return both?

Reply:
To fill in some details from a discussion with @schlafly (thanks!). It sounds like the goal here is to:
- allow a user to call this step and have it just return the catalog
- have the pipeline call this step and update the model passed as input

Having the step return both would mean the step would have to manage the file saving to assign different suffixes (and remove `idx`) for each result (the [ami_analyze step](https://github.com/spacetelescope/jwst/blob/ce9a135f3bc6bba21e95dd47a91fc4d9ba5a1991/jwst/ami/ami_analyze_step.py#L29) in jwst does this).

I'm just leaving this comment for future reference. I initially commented to try and understand the usage (this is the first time I've seen the spec used in this way) and I think now I understand. Thanks and feel free to mark this as "resolved".

Reply:
Not sure as to how `stpipe` will handle multiple returned results.